extensions/ipynb/src/notebookImagePaste.ts

```diff
@@ -66,19 +66,68 @@ class CopyPasteEditProvider implements vscode.DocumentPasteEditProvider {
 
 		return { insertText: pasteSnippet, additionalEdit: workspaceEdit };
 	}
+}
 
-	private getCellFromCellDocument(cellDocument: vscode.TextDocument): vscode.NotebookCell | undefined {
-		for (const notebook of vscode.workspace.notebookDocuments) {
-			if (notebook.uri.path === cellDocument.uri.path) {
-				for (const cell of notebook.getCells()) {
-					if (cell.document === cellDocument) {
-						return cell;
-					}
-				}
-			}
+class ImageDropEditProvider implements vscode.DocumentDropEditProvider {
+	async provideDocumentDropEdits(
+		document: vscode.TextDocument,
+		_position: vscode.Position,
+		dataTransfer: vscode.DataTransfer,
+		_token: vscode.CancellationToken
+	): Promise<vscode.DocumentDropEdit | undefined> {
+		const enabled = vscode.workspace.getConfiguration('ipynb', document).get('experimental.pasteImages.enabled', false);
+		if (!enabled) {
+			return undefined;
+		}
+
+		// get b64 data from paste
+		// TODO: dataTransfer.get() limits to one image dropped
+		const dataItem = dataTransfer.get('image/png');
+		if (!dataItem) {
+			return undefined;
 		}
-		return undefined;
+		const fileDataAsUint8 = await dataItem.asFile()?.data();
+		if (!fileDataAsUint8) {
+			return undefined;
+		}
+
+		// get filename data from paste
```

Comment:
A lot of this logic should be the same as paste. Can you extract this to a shared function both drop and paste can use